packages/netlify-cms-core/src/actions/entries.ts

```diff
@@ -768,8 +777,14 @@ export function persistEntry(collection: Collection) {
         if (assetProxies.length > 0) {
           dispatch(loadMedia());
         }
-        dispatch(entryPersisted(collection, serializedEntry, slug));
-        if (serializedEntry.get('newRecord')) return dispatch(loadEntry(collection, slug));
+        dispatch(entryPersisted(collection, serializedEntry, newSlug));
+        if (collection.has('nested')) {
```

Comment:
similar to editorial workflow, we need to re-load all entries after publish and navigate to the new slug if it had changed